### ISSUE TITLE
Undocumented API change in V5

### ISSUE THREAD

**Issue by Titozzz** [adjust/react_native_sdk#248]
https://github.com/adjust/react_native_sdk/commit/01b06493c5ac9ad319d5a2b2aa59fdca7ce359fc breaks the expectation that the callback will be called (with undefined or more) so it's an undocumented breaking API change.

Example code that code broken:
```tsx
  const updateValues = useCallback(() => {
    Adjust.getAdid((id) => {
      setAdid(id ?? '');
    });
    Adjust.getIdfa((id) => {
      setIdfa(id ?? '');
    });
    Adjust.getGoogleAdId((id) => {
      setGpsAdid(id ?? '');
    });
  }, []);

  useEffect(() => {
    updateValues();
  }, [updateValues]);
.....
  if (adid === undefined || gpsAdid === undefined || idfa === undefined) {
    return null;
  }
....
 return meaningful content
```

**Comment by uerceg** [comment]
Hey @Titozzz,

Thank you for the feedback and definitely a valid point. 👍

When it comes to platform specific methods (including the getters you have mentioned), with v5 we have also made them to run only on the platform they [semantically belong to](https://github.com/adjust/react_native_sdk/blob/master/index.js#L122-L138). But yes, we will document in the migration guide that iOS platform specific getters will no longer return anything if invoked from Android platform and vice versa.

Maybe also a question for you: Do you see any value in getting `undefined` or some invalid value when invoking iOS getters from Android platform or does having the platform specific execution sound better?

**Comment by Titozzz** [comment]
I'm honestly fine with both.

**Comment by AuroPick** [comment]
Yes this should be documented. I was using `requestTrackingAuthorizationWithCompletionHandler` without platform checking. I was continuing app init in handler method.
